mlflow/langchain/__init__.py

```diff
@@ -328,6 +345,27 @@ def log_model(
             type(lc_model.agent.llm_chain.llm).__name__,
         )
 
+    if isinstance(lc_model, langchain.chains.RetrievalQA):
+        if conda_env is None and pip_requirements is None:
+            logger.warning(
+                "RetrievalQA models are currently incompatible with the "
+                "infer_pip_requirements function. Please specify the environment "
+                "requirements in the `conda_env` or `pip_requirements` parameter."
```

Comment:
Why is this incompatible? Assuming we can load the pyfunc model in a subprocess, can't we infer the requirements?

Reply:
We can't load it back from the local temp dir because the loading depends on the fact that the `Retriever` pyfunc model is available in the sub dir, but since `infer_pip_requirements` loads the model from the local temp dir, there is no `Retriever` pyfunc model in the sub dir.

Reply:
Resolved.